spec/shared_contexts.rb

```diff
@@ -0,0 +1,44 @@
+require 'securerandom'
+
+RSpec.shared_context :integration_test_context do
+  subject(:client) do
+    ThreeScale::API.new(endpoint: @endpoint, provider_key: @provider_key, verify_ssl: @verify_ssl)
+  end
+
+  before :context do
+    @endpoint = ENV.fetch('ENDPOINT')
+    @provider_key = ENV.fetch('PROVIDER_KEY')
+    @verify_ssl = !(ENV.fetch('VERIFY_SSL', 'true').to_s =~ /(true|t|yes|y|1)$/i).nil?
+    @apiclient = ThreeScale::API.new(endpoint: @endpoint, provider_key: @provider_key,
+                                     verify_ssl: @verify_ssl)
+    @service_test = @apiclient.create_service('name' => "3scalerubytest#{SecureRandom.uuid}")
```

Comment:
Using instance variables is not very RSpec-y.

Reply:
Those are required objects in `before :context` and `after :context` to ensure objects are all cleaned. Besides, many tests assume one service, one app plan, one app exists. 

Other implementation option would be to use `let(:service)` and then implement `after :each`, but then service, app, app_plan, account would be created and deleted on each iteration.

Reply:
Ok, that is a reasonable compromise 👍